### ISSUE TITLE
Crash when albumless tracks begin playing

### ISSUE THREAD

**Issue by Wevah** [rurza/NepTunes-Issues#64]
When albumless tracks start playing, NepTunes can crash trying to add `nil` to a dictionary.

On macOS 10.15.1.

[NepTunes_2019-11-16-180845_Vergil.crash.zip](https://github.com/rurza/NepTunes-Issues/files/3854899/NepTunes_2019-11-16-180845_Vergil.crash.zip)

**Comment by derekwolpert** [comment]
I've been experiencing intermittent crashes with NepTunes as well. I just attempted to play multiple songs without an album title in Catalina's Music App, and NepTunes crashes reliably each time.

I'm on a 2016 15" MBP running macOS 10.15.2, Music v1.0.2.37, NepTunes 1.70.  Let me know if any other info would be helpful in addressing this issue.

<img width="882" alt="Screen Shot 2019-12-21 at 3 31 15 PM" src="https://user-images.githubusercontent.com/48101033/71313446-fa8a6900-2406-11ea-84a9-64bb81e828b5.png">

**Comment by chriswil** [comment]
Same issue with albumless song.  2020 16" MBP macOS 10.15.7, NepTunes 1.70

mp3: https://phoebebridgers.bandcamp.com/track/iris

**Comment by Wevah** [comment]
I think I've tracked it down to this area in the iTunesSearch CocoaPod starting at line 313:

```objc
    NSDictionary *filters = @{
        @"wrapperType": @"collection",
        @"collectionType": @"album",
        @"collectionName": albumName
    };
```

If `albumName` is `nil`, it'll crash.

A simple fix might be to change the line to

```objc
        @"collectionName": [self forceString:albumName]
```

(in keeping with the use of `forceString:` elsewhere).

I haven't actually built NepTunes myself to test, yet, as I can't remember how to build the pods. 😬

Edit: Right, gotta use the xcworkspace and not the xcproject.

Edit 2: Can't seem to make it crash currently. Will report back if I can repro.

**Comment by rurza** [comment]
I'm alive! :)
Thanks, it'll be fixed with 1.71 release :)

**Comment by derekwolpert** [comment]
@rurza Thanks a bunch!
